File: javascript/2441.largest-positive-integer-that-exists-with-its-negative/2441.largest-positive-integer-that-exists-with-its-negative.js
/*
 * @lc app=leetcode id=2441 lang=javascript
 *
 * [2441] Largest Positive Integer That Exists With Its Negative
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxK = function(nums) {
    let set = new Set();
    let ans = -1;

    for (let i = 0; i < nums.length; i++) {
        if (set.has(nums[i])) {
            if (ans < Math.abs(nums[i])) {
                ans = Math.abs(nums[i])
            }
        } else {
            set.add(-1 * nums[i])
        }
    }

    return ans
};

// @lc code=end

File: rust/src/bin/2441/2441.largest-positive-integer-that-exists-with-its-negative.rs
/*
 * @lc app=leetcode id=2441 lang=rust
 *
 * [2441] Largest Positive Integer That Exists With Its Negative
 */

use std::cmp::max;

// @lc code=start
impl Solution {
    pub fn find_max_k(nums: Vec<i32>) -> i32 {
        let mut map = std::collections::HashSet::<i32>::new();
        let mut ans = -1;

        for i in 0..nums.len() {
            if map.contains(&nums[i]) {
                ans = core::cmp::max(ans, nums[i].abs());
            } else {
                map.insert(-1 * nums[i]);
            }
        }

        ans

    }
}
// @lc code=end
